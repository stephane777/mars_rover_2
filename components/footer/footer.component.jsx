import React from "react";
import sprite from "../../public/images/sprite.svg";
import { PRIMARY } from "../../constants/color";
import { github } from "../../constants/url";

const Footer = () => {
	return (
		<>
			<div className="footer_wrapper">
				<footer>
					<a
						href={github}
						className="footer__link"
						target="_blank"
						rel="noreferrer"
						data-testid="github-link"
					>
						<svg data-testid="github-svg">
							<use
								href={`${sprite}#icon-github`}
								className="grid__icon-use"
							></use>
						</svg>
					</a>
					<p className="createdBy">Created by Stephane Candelas.</p>
				</footer>
			</div>
			<style jsx>{`
				.footer_wrapper {
					height: 10vh;
					display: flex;
					padding-left: 4rem;
					background-color: ${PRIMARY};
					display: flex;
					justify-content: center;
					align-items: center;
				}
				footer {
					display: flex;
					justify-content: center;
					align-items: center;
				}
				svg {
					fill: lightgray;
					width: 2rem;
					height: 2rem;
				}
				.createdBy {
					color: lightgray;
					margin-left: 2rem;
					margin-bottom: 0;
					font-size: 1.2rem;
					font-weight: light;
				}
			`}</style>
		</>
	);
};
export { Footer };
