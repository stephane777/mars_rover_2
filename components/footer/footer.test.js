import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import { Footer } from "./footer.component";

describe("it should render the footer", () => {
	it("should render the footer", () => {
		const { container } = render(<Footer />);
		expect(container.firstChild.className).toMatch("footer_wrapper");
	});
	it("should show Created by ...", () => {
		render(<Footer />);
		expect(
			screen.getByText(/Created by Stephane Candelas./i)
		).toBeInTheDocument();
	});
	it("should contain a link", () => {
		render(<Footer />);
		expect(screen.queryByTestId("github-link")).toBeTruthy();
		expect(screen.queryByTestId("github-svg")).toBeTruthy();
	});
});

describe("SNAPSHOT", () => {
	it("should match the snapshot for Footer component", () => {
		const snapshot = render(<Footer />);
		expect(snapshot).toMatchSnapshot();
	});
});
