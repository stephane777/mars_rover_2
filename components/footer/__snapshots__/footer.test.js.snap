// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOT should match the snapshot for Footer component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="jsx-1760820722 footer_wrapper"
      >
        <footer
          class="jsx-1760820722"
        >
          <a
            class="jsx-1760820722 footer__link"
            data-testid="github-link"
            href="https://github.com/stephane777"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              class="jsx-1760820722"
              data-testid="github-svg"
            >
              <use
                class="jsx-1760820722 grid__icon-use"
                href="#icon-github"
              />
            </svg>
          </a>
          <p
            class="jsx-1760820722 createdBy"
          >
            Created by Stephane Candelas.
          </p>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="jsx-1760820722 footer_wrapper"
    >
      <footer
        class="jsx-1760820722"
      >
        <a
          class="jsx-1760820722 footer__link"
          data-testid="github-link"
          href="https://github.com/stephane777"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="jsx-1760820722"
            data-testid="github-svg"
          >
            <use
              class="jsx-1760820722 grid__icon-use"
              href="#icon-github"
            />
          </svg>
        </a>
        <p
          class="jsx-1760820722 createdBy"
        >
          Created by Stephane Candelas.
        </p>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
